internal/server/deviceflow.go

```diff
@@ -112,7 +112,7 @@ func (a *API) ApproveDeviceAdd(c *gin.Context, req *api.ApproveDeviceFlowRequest
 		return nil, internal.ErrExpired
 	}
 
-	if dfar.AccessKey != nil {
+	if dfar.AccessKeyID != 0 {
```

Comment:
This fixes a bug. `AccessKey` is always nil because we don't populate it in the query used above.